src/core/thread/mle.cpp

```diff
@@ -2089,7 +2093,8 @@ otError Mle::SendChildUpdateRequest(void)
 
     if (mChildUpdateAttempts >= kMaxChildKeepAliveAttempts)
     {
-        mChildUpdateAttempts = 0;
+        mChildUpdateAttempts     = 0;
+        mChildUpdateRequestState = kChildUpdateRequestNone;
         BecomeDetached();
         ExitNow();
     }
```

Comment:
How about setting `mChildUpdateRequest = kChildUpdateRequestActive` after this `if`?

I think this change can fix an bug we currently have in the code where on SED, "Child Update" is attempted once and after the timeout, a "Data Request" is sent (please see current [`HandleChildUpdateRequestTimer`](https://github.com/openthread/openthread/blob/f0654381117d6f515e5808a1750761fe2719c5a4/src/core/thread/mle.cpp#L2064-L2081)).

If we change state to `Active`, when other modules call `Mle::SendChildUpdateRequest()`, if no response is received after timeout we will know to send a "Child Update".

Reply:
Agree with the proposal. will update as your suggestion.